src/ja/common/message/base.py

```diff
@@ -3,33 +3,97 @@
 implemented in order for a class to be transferable between the components
 of JobAdder.
 """
-from abc import ABC
+from abc import ABC, abstractmethod
+from typing import Dict
 
 
 class Serializable(ABC):
     """
-    A base class for all objects which can be converted to and from a python
+    A base class for all objects which can be converted to and from a Python
     dictionary. The dictionary should contain only basic python types:
-    strings, ints, bytes, doubles, lists and dictionaries.
+    strings, ints, bytes, doubles, lists and dictionaries. The dictionary can
+    also be dumped as/read from a YAML string.
     """
 
+    @abstractmethod
+    def to_dict(self) -> Dict[str, object]:
+        """!
+        @return A dictionary representation of this object.The entries in the
+        dictionary are equivalent to the properties of this object.
+        """
 
-class Message(Serializable):
+    @classmethod
+    @abstractmethod
+    def from_dict(cls, property_dict: Dict[str, object]) -> "Serializable":
+        """!
+        @param property_dict A Python dictionary defining the desired
+        properties of the Serializable object to be created.
+        @return A new Serializable object based on the property entries of the
+        specified dictionary.
+        """
+
+    @abstractmethod
```

Comment:
Why is this and the `from_string` an abstract method? If we have conversion to dictionary, we can just use this conversion to stringify and convert back again.

Reply:
You're right.